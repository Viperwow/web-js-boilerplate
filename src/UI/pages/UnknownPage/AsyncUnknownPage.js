import loadable from 'src/common/loadable';
import Loader from 'src/UI/components/Loader';

const DynamicUnknownPage = loadable({
  importer: () => import('src/UI/pages/UnknownPage/UnknownPageContainer' /* webpackChunkName: "UnknownPage" */), // eslint-disable-line max-len
  LoaderComponent: Loader,
});

export default DynamicUnknownPage;
