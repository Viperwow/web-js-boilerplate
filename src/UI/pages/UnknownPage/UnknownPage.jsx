import {identity as _identity} from 'lodash';
import React from 'react';
import {Link} from 'react-router-dom';

const UnknownPage = ({
  text,
  t = _identity,
}) => (
  <div className="page">
    {t(text)}
    <ul>
      <li>
        <Link to="/1" replace>
          1
        </Link>
      </li>
      <li>
        <Link to="/2" replace>
          2
        </Link>
      </li>
      <li>
        <Link to="/3" replace>
          3
        </Link>
      </li>
    </ul>
  </div>
);

export default UnknownPage;
