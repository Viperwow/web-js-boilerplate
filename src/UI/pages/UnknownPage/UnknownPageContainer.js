import connectQuery from 'src/common/connectQuery';
import withOnError from 'src/common/withOnError';
import withTranslate from 'src/common/withTranslate';
import {compose} from 'src/helpers/utility';
import {LOCALE_QUERY} from 'src/queries';
import UnknownPage from './UnknownPage';

const mapStateToProps = (_, {t}) => ({
  text: t('pages.unknown'),
});

export default compose(
  withTranslate,
  connectQuery(LOCALE_QUERY, mapStateToProps),
  withOnError,
)(UnknownPage);
