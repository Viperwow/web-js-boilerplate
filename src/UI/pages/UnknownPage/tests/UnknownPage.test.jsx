import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import UnknownPage from '../UnknownPage';

jest.unmock('../UnknownPage');

describe('UnknownPage component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<UnknownPage />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });

  test('With text prop', () => {
    const output = shallow(<UnknownPage text="text" />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
