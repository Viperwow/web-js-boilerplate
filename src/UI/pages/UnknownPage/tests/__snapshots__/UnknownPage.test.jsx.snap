// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownPage component should render correctly With default props 1`] = `
<div>
  <ul>
    <li>
      <Link
        replace={true}
        to="/1"
      >
        1
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/2"
      >
        2
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/3"
      >
        3
      </Link>
    </li>
  </ul>
</div>
`;

exports[`UnknownPage component should render correctly With text prop 1`] = `
<div>
  text
  <ul>
    <li>
      <Link
        replace={true}
        to="/1"
      >
        1
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/2"
      >
        2
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/3"
      >
        3
      </Link>
    </li>
  </ul>
</div>
`;
